pkg/controller/sync/sync_controller.go

```diff
@@ -56,6 +56,11 @@ func (a *Adder) Add(mgr manager.Manager) error {
 		log.Error(err, "Sync metrics reporter could not start")
 		return err
 	}
+	err = reporter.RegisterCallback(a.CacheManager.GetSyncMetricsCache())
```

Comment:
controller should not be responsible for callback registration. Also (and more importantly), sharing state like this goes against the rest of the pre-existing `_reporter.go` design.